app/src/main/java/com/github/libretube/services/DownloadService.kt

```diff
@@ -1,182 +1,524 @@
+            .addAction(getPauseAction(item.id))
 
-        with(NotificationManagerCompat.from(this)) {
-            // notificationId is a unique int for each notification that you must define
-            notify(DOWNLOAD_FAILURE_NOTIFICATION_ID, builder.build())
+        notificationManager.notify(item.getNotificationId(), notificationBuilder.build())
+    }
+
+    private fun setPauseNotification(
+        notificationBuilder: NotificationCompat.Builder,
+        item: DownloadItem,
+        isCompleted: Boolean = false
+    ) {
+        notificationBuilder
+            .setProgress(0, 0, false)
+            .setOngoing(false)
+            .clearActions()
+
+        if (isCompleted) {
+            notificationBuilder
+                .setSmallIcon(R.drawable.ic_done)
+                .setContentText(getString(R.string.download_completed))
+        } else {
+            notificationBuilder
+                .setSmallIcon(R.drawable.ic_pause)
+                .setContentText(getString(R.string.download_paused))
+                .addAction(getResumeAction(item.id))
         }
+        notificationManager.notify(item.getNotificationId(), notificationBuilder.build())
     }
 
-    private fun downloadSucceededNotification() {
-        Log.i(TAG(), "Download succeeded")
-        val builder = NotificationCompat.Builder(this, DOWNLOAD_CHANNEL_ID)
-            .setSmallIcon(R.drawable.ic_download)
-            .setContentTitle(resources.getString(R.string.success))
-            .setContentText(getString(R.string.downloadsucceeded))
-            .setPriority(NotificationCompat.PRIORITY_HIGH)
+    private fun getResumeAction(id: Int): NotificationCompat.Action {
+        val intent = Intent(this, NotificationReceiver::class.java)
+
+        intent.action = ACTION_DOWNLOAD_RESUME
+        intent.putExtra("id", id)
 
-        with(NotificationManagerCompat.from(this)) {
-            // notificationId is a unique int for each notification that you must define
-            notify(DOWNLOAD_SUCCESS_NOTIFICATION_ID, builder.build())
+        val flags = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            (PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
+        } else {
+            PendingIntent.FLAG_UPDATE_CURRENT
         }
+
+        return NotificationCompat.Action.Builder(
+            R.drawable.ic_play,
+            getString(R.string.resume),
+            PendingIntent.getBroadcast(this, id, intent, flags)
+        ).build()
     }
 
-    override fun onDestroy() {
-        try {
-            unregisterReceiver(onDownloadComplete)
-        } catch (e: Exception) {
-            e.printStackTrace()
+    private fun getPauseAction(id: Int): NotificationCompat.Action {
+        val intent = Intent(this, NotificationReceiver::class.java)
+
+        intent.action = ACTION_DOWNLOAD_PAUSE
+        intent.putExtra("id", id)
+
+        val flags = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
That could be moved into a function called `getNotificationFlags()` as its used for the other  notification at `getResumeAction()` too.